datalad/distribution/metadata.py

```diff
@@ -0,0 +1,116 @@
+      -> "dc:contributor.author#4"
+    - by term/value pair: -> "dc:contributor.author#Michael"
+    """
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to perform the install operation on.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory and/or the `path` given""",
+            constraints=EnsureDataset() | EnsureNone()),
+        add=Parameter(
+            args=("--add",),
+            action="append",
+            nargs=2,
+            metavar="VALUE",
+            doc="""[PY: list of PY] term/value pair [PY: tuples PY] to be added
+            to the metadata.[CMD:  This option can be given multiple times.
+            CMD]"""),
+        get=Parameter(
+            args=("--get",),
+            metavar="TERM",
+            nargs='+',
+            doc="""[PY: list of PY] terms to be reported from the metadata. For
+            the supported syntax to address a term, see the general
+            documentation"""),
+        remove=Parameter(
+            args=("--remove",),
+            metavar="TERM",
+            nargs='+',
+            doc="""[PY: list of PY] terms to be reported from the metadata. For
+            the supported syntax to address a term, see the general
+            documentation"""),
+        replace=Parameter(
+            args=("--replace",),
+            action="append",
+            nargs=2,
+            metavar="VALUE",
+            doc="""[PY: list of PY] term/value pair [PY: tuples PY] identifying
+            the to-be-replaced item by its term (first value), and the
+            replacement value (second value).[CMD:  This option can be given
+            multiple times. CMD] For the supported syntax to address a term,
+            see the general documentation"""),
+        category=Parameter(
+            args=("--category",),
+            metavar="LABEL",
+            # might think of special label "all"...
+            doc="""name of the metadata category to operate on."""),
+        import_from=Parameter(
+            args=("--import-from",),
+            metavar="SCHEMALABEL",
+            doc="""label of a supported metadata scheme to import metadata
+            from. Any existing metadata is replaced with the imported
+            information. Use in conjunction with the metadata `category`
+            setting to manage multiple sets of metadata without impacting
+            the main (possible hand-curated) metadata."""),
+        aggregate=Parameter(
+            args=("--aggregate",),
+            action="store_true",
+            doc="""aggregate metadata from any installed subdataset. If a
+            recursion limit is set, installed subdataset exceeding this
+            limit will be ignored. Aggregation honors the selected metadata
+            category"""),
+        recursion_limit=recursion_limit,
```

Comment:
I wondered if we need an explicit `--recursive` to signal traversing the entire tree.  E.g. if we have ds1/ds2/ds3/ds4 hierarchy, and they all set (we need a persistent configuration setting for that) to collect meta only from direct child, but we want to "update" (i.e. aggregate) in all of them, while invoking this command from within ds1...

or may be such type of recursion could become a common option for more commands?  e.g. "save", and even "get" (after we introduce ability to request content e.g. by tags) could all have the same handling of such recursion -- just go from the leaves up (depth first)